// mat-sidenav-menu/esm5/lib/item.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';
var SideNavItemComponent = /** @class */ (function () {
    function SideNavItemComponent(router) {
        this.router = router;
    }
    /**
     * @return {?}
     */
    SideNavItemComponent.prototype.navigate = /**
     * @return {?}
     */
    function () {
        if (!this.disabled) {
            this.router.navigate([this.route]);
        }
    };
    /**
     * @param {?} path
     * @return {?}
     */
    SideNavItemComponent.prototype.activatePath = /**
     * @param {?} path
     * @return {?}
     */
    function (path) {
        if (path === this.route) {
            this.active = true;
        }
        else {
            this.active = false;
        }
    };
    SideNavItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'side-nav-item',
                    template: "<li>\n                  <a class=\"side-nav-item\" matRipple\n                    [ngClass]=\"{'side-nav-item-active': active,'side-nav-item-disabled': disabled}\"\n                    (click)=\"navigate()\">\n                        <mat-icon *ngIf=\"icon\" style=\"margin-right:10px;vertical-align:middle\">{{ icon }}</mat-icon>\n                        <span style=\"vertical-align:middle\">{{ title }}</span>\n                  </a>\n               </li>",
                    styles: [
                        ".side-nav-item {\n          list-style-type: none;\n          box-sizing: border-box;\n          display: block;\n          font-size: 14px;\n          font-weight: 400;\n          line-height: 47px;\n          text-decoration: none;\n          transition: all .3s;\n          padding: 0 16px;\n          position: relative;\n          cursor: pointer;\n        }"
                    ]
                },] },
    ];
    /** @nocollapse */
    SideNavItemComponent.ctorParameters = function () { return [
        { type: Router, },
    ]; };
    SideNavItemComponent.propDecorators = {
        "title": [{ type: Input, args: ['title',] },],
        "icon": [{ type: Input, args: ['icon',] },],
        "route": [{ type: Input, args: ['route',] },],
        "disabled": [{ type: Input, args: ['disabled',] },],
    };
    return SideNavItemComponent;
}());
export { SideNavItemComponent };
function SideNavItemComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    SideNavItemComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    SideNavItemComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    SideNavItemComponent.propDecorators;
    /** @type {?} */
    SideNavItemComponent.prototype.title;
    /** @type {?} */
    SideNavItemComponent.prototype.icon;
    /** @type {?} */
    SideNavItemComponent.prototype.route;
    /** @type {?} */
    SideNavItemComponent.prototype.disabled;
    /** @type {?} */
    SideNavItemComponent.prototype.active;
    /** @type {?} */
    SideNavItemComponent.prototype.router;
}
