// mat-sidenav-menu/esm5/lib/mat-sidenav-menu.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Component, ContentChildren, Input, QueryList } from '@angular/core';
import { Router, ActivatedRoute, NavigationEnd } from '@angular/router';
import { MatSidenav } from '@angular/material';
import { SideNavItemComponent } from './item';
var MatSidenavMenuComponent = /** @class */ (function () {
    function MatSidenavMenuComponent(router, aroute) {
        var _this = this;
        this.router = router;
        this.aroute = aroute;
        // console.log('router:', this.router);
        this.router.events.subscribe(function (e) {
            // console.log('url:', this.router.url);
            // console.log('router state:', this.router.routerState);
            // if (this.router.routerState.root.children.length > 0 ) {
            if (e instanceof NavigationEnd) {
                // console.log('component:', '(' + Date.now + ')' + this.router.routerState.root.children[0].component);
                // console.log('path: ' + this.getPath());
                // console.log('component:', '(' + Date.now + ')' + this.router.routerState.root.children[0].component);
                // console.log('path: ' + this.getPath());
                _this.items.forEach(function (i) {
                    i.activatePath(_this.getPath());
                });
                _this.sidenav.close();
            }
            // console.log('path:', this.erouterState);
        });
        // this.aroute.url.subscribe( (u) => {
        //   console.log('active route:', u);
        // });
    }
    /**
     * @return {?}
     */
    MatSidenavMenuComponent.prototype.getPath = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ path = this.router.routerState.snapshot.url;
        return path;
    };
    MatSidenavMenuComponent.decorators = [
        { type: Component, args: [{
                    selector: 'side-nav-menu',
                    template: "\n      <nav style=\"height:100%\">\n        <mat-toolbar color=\"primary\">\n          <button mat-icon-button (click)=\"sidenav.close()\"><mat-icon>keyboard_arrow_left</mat-icon></button>&nbsp;\n          <span>{{ title }}</span>\n        </mat-toolbar>\n\n        <ul class=\"side-nav-menu\">\n          <ng-content></ng-content>\n        </ul>\n      </nav>",
                    styles: [
                        ".side-nav-menu {\n          list-style-type: none;\n          padding: 0\n      }"
                    ]
                },] },
    ];
    /** @nocollapse */
    MatSidenavMenuComponent.ctorParameters = function () { return [
        { type: Router, },
        { type: ActivatedRoute, },
    ]; };
    MatSidenavMenuComponent.propDecorators = {
        "sidenav": [{ type: Input },],
        "title": [{ type: Input },],
        "items": [{ type: ContentChildren, args: [SideNavItemComponent, { descendants: true },] },],
    };
    return MatSidenavMenuComponent;
}());
export { MatSidenavMenuComponent };
function MatSidenavMenuComponent_tsickle_Closure_declarations() {
    /** @type {!Array<{type: !Function, args: (undefined|!Array<?>)}>} */
    MatSidenavMenuComponent.decorators;
    /**
     * @nocollapse
     * @type {function(): !Array<(null|{type: ?, decorators: (undefined|!Array<{type: !Function, args: (undefined|!Array<?>)}>)})>}
     */
    MatSidenavMenuComponent.ctorParameters;
    /** @type {!Object<string,!Array<{type: !Function, args: (undefined|!Array<?>)}>>} */
    MatSidenavMenuComponent.propDecorators;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.sidenav;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.title;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.items;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.active;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.router;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.aroute;
}
